### PR TITLE
Push notifications integration tests for Android

### PR DESCRIPTION
**Pull request by b1zzu**
This was the old PR: https://github.com/jhellar/aerogear-integration-tests/pull/1

### CODE REVIEW

**Comment by b1zzu** [comment]
I've split metrics and ups database again because the metrics service will start before the metrics db is created in postgresql causing the metrics service to fail, and depends_on will not working correctly because it does work only when using env and not .sql

**Comment by psturc** [comment]
👀

File: docker-compose.yml [hunk 1]
```diff
@@ -10,37 +10,44 @@ services:
...
     image: postgres:9.6
-    container_name: postgres
-    volumes:
-    - "./config/postgres.sql:/docker-entrypoint-initdb.d/postgres.sql"
-    ports:
-      - "5432:5432"
+    container_name: metricsdb
+    environment:
+      POSTGRES_USER: metrics	
+      POSTGRES_PASSWORD: metrics
```
Comment (psturc) [inline docker-compose.yml]: not sure if it's my `docker-compose` version, but these tabs make the command to fail:
```
docker-compose up -d
ERROR: yaml.scanner.ScannerError: while scanning for the next token
found character '\t' that cannot start any token
  in "./docker-compose.yml", line 17, column 29
```
lines: 17, 18 & 36

File: Jenkinsfile [hunk 2]
```diff
@@ -30,7 +30,9 @@ node('psi_rhel8') {
           credentialsId: 'browserstack',
           usernameVariable: 'BROWSERSTACK_USER',
           passwordVariable: 'BROWSERSTACK_KEY'
-        )
+        ),
+        string(credentialsId: 'firebase-server-key', variable: 'FIREBASE_SERVER_KEY'),
+        string(credentialsId: 'firebase-sender-id', variable: 'FIREBASE_SENDER_ID'),
```
Comment (psturc) [inline Jenkinsfile]: Should we update readme with these variables as a prerequisite when someone is running the tests locally?

File: Jenkinsfile [hunk 3]
```diff
@@ -78,8 +80,11 @@ node('psi_rhel8') {
                 sh 'npm set registry http://verdaccio:4873/'
                 sh 'apt install gradle'
                 sh 'npm -g install cordova@8'
-                git branch: 'master', url: 'https://github.com/jhellar/aerogear-integration-tests.git'
-                sh './scripts/build-testing-app.sh'
+                checkout scm
+                withCredentials([file(credentialsId: 'google-services', variable: 'GOOGLE_SERVICES')]) {
```
Comment (psturc) [inline Jenkinsfile]: same here

**Comment by psturc** [comment]
@b1zzu is there anything else I need to set up to run the tests locally?
I followed the instructions in readme and the additional variables mentioned above in comments.

But when I run the tests I'm getting
```
  1) "before all" hook: connect to postgres in "{root}":
     Error: connect ECONNREFUSED 127.0.0.1:5432
      at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1191:14)
```

File: Jenkinsfile (hunk 2 above)
Comment (b1zzu) [reply Jenkinsfile]: yes definetly, maybe it is also good to verify that these variables are set before starting the tests

**Comment by b1zzu** [comment]
> @b1zzu is there anything else I need to set up to run the tests locally?
> I followed the instructions in readme and the additional variables mentioned above in comments.
> 
> But when I run the tests I'm getting
> 
> ```
>   1) "before all" hook: connect to postgres in "{root}":
>      Error: connect ECONNREFUSED 127.0.0.1:5432
>       at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1191:14)
> ```

@psturc this is caused by some of my latest changes that I still have to finish, I ping you again when I have finished them

**Comment by b1zzu** [comment]
@psturc I've updated the documentation and fixed the issue with local tests, you can try it again now

**Comment by psturc** [comment]
All tests except `push` were working fine for me locally.

Maybe I messed up my local configuration somehow, so I will give it a try later again on a freshly cloned repo. I checked the logs on browserstack, but it didn't help me much to find what's wrong there.

The error was:
```
  Push
2019-06-11T12:20:03.356Z ERROR webdriver: Request failed due to Error: script timeout: result was not received in 30 seconds
    at getErrorFromResponseBody (/Users/psturc/work/aerogear/aerogear-integration-tests/node_modules/webdriver/build/utils.js:350:10)
```

**Pull request merged**